### DISCUSSION TITLE
GIF sending is not working

### DISCUSSION THREAD

**Discussion by mircoianese** [tulir/whatsmeow#696] [General]
Hello, I'm new to the library.

I have this code to send a GIF in a newsletter:

```
data, err := os.ReadFile("./test-gif.gif")
if err != nil {
  fmt.Println("Failed to load file", err)
  return
}

uploaded, err := client.UploadNewsletter(context.Background(), data, whatsmeow.MediaVideo)
if err != nil {
  fmt.Println("Failed to upload file", err)
  return
}

message := &waE2E.Message{
   VideoMessage: &waE2E.VideoMessage{
	Caption:    proto.String("Hello"),
	Mimetype:   proto.String("image/gif"),
	URL:        &uploaded.URL,
	DirectPath: &uploaded.DirectPath,
	MediaKey:   uploaded.MediaKey,
	//FileEncSHA256: uploaded.FileEncSHA256,
	FileSHA256:    uploaded.FileSHA256,
	FileLength:    &uploaded.FileLength,
	GifPlayback:   proto.Bool(true),
	//JPEGThumbnail: thumbnailBuf.Bytes(), // removed for this example
  },
}

client.SendMessage(context.Background(), newsletter.ID, message)
```

The GIF is a valid GIF file and it get's uploaded correctly: https://mmg.whatsapp.net/m1/v/t24/An-wvUPHyVUset78fPewM3sj2tYtAo68-jZnEpEk1g5Z1J4uSU1OwfG2qN2tz7eZ4h64IBdpbI-YLIfBUZOJvXGuJbmBcNe2QC6IC3wJsYqiq3_QiSgM8Vtnsphl3mU2bgcr?ccb=10-5&oh=01_Q5AaIJEqjeAUk2y9KG_7I98jUm75GsVwpsKLMphLYFWNXJ7Q&oe=675E965E&_nc_sid=5e03e0&mms3=true

But on the newsletter when I try to open it I get a black screen (only the thumbnail is shown if specified in the request). 
Any idea on what's wrong here?

Thank you!

**Comment by ghost** [comment]
Actually, gif message is a video file. `video/mp4`

**Reply by mircoianese** [reply]
Thank you for the answer! Unfortunately it's still not working. Do I need to convert the GIF to mp4?

**Reply by ghost** [reply]
Yes, that is.

**Reply by mircoianese** [reply]
I confirm it's working. Thanks
